HTI_2.0.md

```diff
@@ -0,0 +1,451 @@
+* The contents of the message can no longer be inspected by the end-user.
+* The message can only be unpacked by the receiver
+
+This adds an extra layer of security over the communication, minimizing the risk of data leak even further.
+
+Disadvantages of using JWE encryption are:
+* Added complexity to the configuration and message creation.
+* Increased difficulty to debug the contents of the JWT message.
+
+#### Restrictions
+The additional requirement for using a JWE message is as follows:
+* The header must contain the key id (kid) of the public key used for encryption.
+
+#### Layout of the message
+For details of the JWE standard, we refer to the JWE documentation. The diagram below displays the JWT and JWE token from a conceptual perspective
+
+![](images/image7.png)
+
+#### Tip: JWT and JWE message detection
+If you, as a module provider, wish to support both JWE and JWT tokens, it is possible to detect id the token is a JWT or JWE token by counting the number of dots in the token. The following rules apply.
+* If the token contains two dots (.), the token is a JWT token.
+* If the token contains four dots (.), the token is a JWT token wrapped in a JWE token.
+
+#### JWE configuration and storage requirements
+The portal needs to configure the following:
+* The public key of the module provider message encryption. The key id (kid) must be part of the token header.
+
+The module provider needs to configure the following
+* The private key for message decryption, referred to by the kid in the message header. The public key of the keypair should be communicated with the module provider for encryption.
+
+![](images/image3.png)
+
+# Implementation checklist
+This section provides an overview of the requirements and responsibilities in Moscow statements of both the portal and module application.
+
+## The portal application
+
+| Description |
+| ------------- |
+| The HTI claims MUST only contain information about the functional task, the definition of the task, and the people involved. |
+| The JWT message MUST only contain information about the sending system, the recipient system and the message itself, with exception to the `sub` field. |
+| The exchange of the message MUST NOT contain any information about: |
+| * the functional task, the definition of the task, and the people involved, and |
+| * information about the sending system, the recipient system and the message itself. |
+| The fields used in the HTI claims object MUST match the table HTI claims field mapping. |
+| The resource field MUST be persistent over the timeframe the task is active. |
+| The definition reference MUST be an URI |
+ | The user identifier (`sub` and `patient`) MUST be a persistent pseudo identifier. |
+ | The user identifier (`sub` and `patient`) MUST be both unique and persistent in each domain |
+ | The user identifier (`sub` field) MUST be used for the user that should execute the task. |
```

Comment:
`sub` zal in veel gevallen juist niet de taak uitvoeren maar beoordelen. Wordt hier `patient` bedoeld?